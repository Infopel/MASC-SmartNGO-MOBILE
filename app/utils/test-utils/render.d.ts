export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
