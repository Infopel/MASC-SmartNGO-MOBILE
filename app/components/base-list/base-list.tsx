import { Placeholder } from "components/placeholder/placeholder"
import { Divider, FlatList } from "native-base"
import { IFlatListProps } from "native-base/lib/typescript/components/basic/FlatList"
import * as React from "react"

export interface BaseListProps<T> extends IFlatListProps<T> {

}

/**
 * Describe your component here
 */
export function BaseList<T>({ ...rest }: BaseListProps<T>) {


  return (
    <FlatList ItemSeparatorComponent={() => <Divider mb='2' />}  ListEmptyComponent={<Placeholder />} {...rest} />
  )
}
