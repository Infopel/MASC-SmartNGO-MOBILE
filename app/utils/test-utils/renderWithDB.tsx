jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  