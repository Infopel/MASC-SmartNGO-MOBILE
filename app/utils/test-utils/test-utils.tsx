import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

