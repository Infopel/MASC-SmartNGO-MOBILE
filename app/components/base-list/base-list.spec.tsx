import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { BaseList } from "./base-list"



describe("<BaseList>",()=>{
  it("render", async function () {
    const component = <BaseList></BaseList>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
