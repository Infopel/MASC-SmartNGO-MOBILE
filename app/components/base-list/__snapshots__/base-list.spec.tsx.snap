// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseList> render 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={<Placeholder />}
  contentContainerStyle={
    Array [
      Object {},
      Object {
        "dataSet": Object {},
      },
      Object {},
    ]
  }
  dataSet={Object {}}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      style={
        Object {
          "padding": 16,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Empty List
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
